src/remote-data.ts

```diff
@@ -787,30 +815,43 @@ const ap = <L, A, B>(fab: RemoteData<L, Function1<A, B>>, fa: RemoteData<L, A>):
 const map = <L, A, B>(fa: RemoteData<L, A>, f: Function1<A, B>): RemoteData<L, B> => fa.map(f);
 const chain = <L, A, B>(fa: RemoteData<L, A>, f: Function1<A, RemoteData<L, B>>): RemoteData<L, B> => fa.chain(f);
 
-//Foldable
+//Foldable2v
 const reduce = <L, A, B>(fa: RemoteData<L, A>, b: B, f: Function2<B, A, B>): B => fa.reduce(f, b);
+const foldMap = <M>(M: Monoid<M>) => <L, A>(fa: RemoteData<L, A>, f: (a: A) => M): M =>
+	fa.isSuccess() ? f(fa.value) : M.empty;
+const foldr = <L, A, B>(fa: RemoteData<L, A>, b: B, f: (a: A, b: B) => B): B => (fa.isSuccess() ? f(fa.value, b) : b);
+
+//Traversable2v
+const traverse = <F>(F: Applicative<F>) => <L, A, B>(
+	ta: RemoteData<L, A>,
+	f: (a: A) => HKT<F, B>,
+): HKT<F, RemoteData<L, B>> => {
+	if (ta.isSuccess()) {
+		return F.map<B, RemoteData<L, B>>(f(ta.value), of);
+	} else if (ta.isFailure()) {
+		return F.of((ta as unknown) as RemoteFailure<L, B>);
+	} else if (ta.isInitial()) {
+		return F.of((ta as unknown) as RemoteInitial<L, B>);
+	} else {
+		return F.of((ta as unknown) as RemotePending<L, B>);
+	}
+};
+const sequence = <F>(F: Applicative<F>) => <L, A>(ta: RemoteData<L, HKT<F, A>>): HKT<F, RemoteData<L, A>> => {
```

Comment:
What do you think of implementing `sequence` through the `traverse` with `identity`?

Reply:
We could do that. You mean like this?

```ts
const sequence = <F>(F: Applicative<F>) => <L, A>(ta: RemoteData<L, HKT<F, A>>): HKT<F, RemoteData<L, A>> =>
	traverse(F)(ta, identity);
```